scripts/ci/check-tutorials-overview-page.py

```diff
@@ -0,0 +1,77 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Check all tutorials are listed in the overview page once in each section.
+
+This splits the page by second-level heading (`##`), then uses a regex to detect
+links of the form `* [Title](/path)`.
+"""
+
+from pathlib import Path
+from dataclasses import dataclass
+import re
+import sys
+
+TUTORIALS_ROOT = Path("./docs/tutorials/")
+OVERVIEW_PAGE = Path(TUTORIALS_ROOT, "index.mdx")
+
+# All links must be of the form
+# * [Tutorial title](/tutorials/tutorial-slug)
+LINK_REGEX = r"\*\s+\[.*\]\((/tutorials\/.*)\)"
+
+
+@dataclass
+class OverviewSection:
+    title: str
+    paths: list[str]
+
+    @classmethod
+    def from_mdx(cls, mdx: str):
+        section_title = mdx.split("\n", 1)[0].strip()
+        return OverviewSection(
+            title=section_title,
+            paths=re.findall(LINK_REGEX, mdx),
+        )
+
+
+if __name__ == "__main__":
+    all_tutorial_paths = [
```

Comment:
Can you please make all the contents of this main guard be in a function called `main()`? It is more idiomatic. Then this if statement only calls `main()`